src/components/directory-content.js

```diff
@@ -774,29 +787,44 @@ const DirectoryContent = () => {
         let children = getSelectedChildren();
         return (
             children.length === 1 &&
-            children[0].type === elementType.FILTERS_CONTINGENCY_LIST
+            children[0].type === elementType.CONTINGENCY_LIST &&
+            childrenMetadata[children[0].elementUuid].subtype ===
+                contingencyListSubtype.FILTERS
         );
     };
 
     const allowsReplaceContingencyWithScript = () => {
         let children = getSelectedChildren();
         return (
             children.length === 1 &&
-            children[0].type === elementType.FILTERS_CONTINGENCY_LIST &&
+            children[0].type === elementType.CONTINGENCY_LIST &&
+            childrenMetadata[children[0].elementUuid].subtype ===
```

Comment:
same, must check if `childrenMetadata[children[0].elementUuid]` is OK first

Reply:
done